mbuild/lattice.py

```diff
@@ -0,0 +1,260 @@
+            elif dimension == 2:
+                basis_vectors['default'].append(([0, 0]))
+            else:
+                basis_vectors['default'].append(([0]))
+        elif (isinstance(basis_vectors, list) or
+                isinstance(basis_vectors, tuple)):
+            for lst in basis_vectors:
+                if len(lst) != 2:
+                    ValueError('Too many arguments per basis vector. '
+                               '{} was provided, but the format should be: '
+                               ' vector = ( (\'ID1\' , [x1,y1,z1]), '
+                               '( \'ID2\' , [x2,y2,z2]) )'.format(lst))
+                # input check ensure ID is a string, and atom location is list
+                elif not isinstance(lst[0], str):
+                    TypeError('ID Error, not a string. {} is not of '
+                              'type str. Please correct in the '
+                              'basis_vectors input {}.'
+                              .format(lst[0], basis_vectors))
+                else:
+                    pass
+
+                if len(lst[1]) == dimension:
+                    if (isinstance(lst[1], tuple) or isinstance(lst[1], list)):
+                        for i in range(dimension):
+                            if (lst[1][i] < 0 or lst[1][i] >= 1):
+                                ValueError('Incorrect basis fractions. '
+                                           '{} is supposed be 0 >= and < 1.'
+                                           .format(lst[1]))
+                            else:
+                                pass
+
+                    else:
+                        TypeError('Incorrect Format: expected <list> '
+                                  'or <tuple>. {} was passed in as type: {} '
+                                  'from basis_vectors.'
+                                  .format(lst[1], type(lst[1])))
+                else:
+                    ValueError('Basis Vector does not match dimension. '
+                               '{} does not have vector components equal to '
+                               'the dimension specified: {}'
+                               .format(lst[1], dimension))
+        else:
+            TypeError('Incorrect Type: "basis_vectors" is of type {}, '
+                      'not type <list> or <tuple>. Please review '
+                      'the documentation for proper format.'
+                      .format(type(basis_vectors)))
+
+        # basis overlap check
+        """
+        check for two same basis values in dict already
+        then check if 2 vectors then overlap
+        error if that occurs
+        Achieved by expanding unit cell out 2 iterations
+        """
+        if isinstance(basis_vectors, defaultdict):
+            pass
+        else:
+            # test for overlap
+            overlap_dict = defaultdict(list)
+            for the_id, vector in basis_vectors:
+                if dimension == 3:
```

Comment:
You should be able to combine these 3 if statements by looping over `x in range(dimension)`